src/kernel/include/arch/riscv64/irq.h

```diff
@@ -0,0 +1,40 @@
+#ifndef UKO_OS_KERNEL__IRQ_H
```

Comment:
oh, didn't realize this was going to so closely correspond to what's in hartlock.c -- one thing you can do is to have an arch-independent `foo.h`, but put the `.c` under arch/